### PR TITLE
Added support for tables in split_multi_hts

### PR DESCRIPTION
**Pull request by konradjk**

### CODE REVIEW

File: python/hail/methods/statgen.py [hunk 1]
```diff
@@ -1463,11 +1463,11 @@ def make_array(cond):
     return left.union(moved) if is_table else left.union_rows(moved)
 
 
-@typecheck(ds=MatrixTable,
+@typecheck(ds=oneof(Table, MatrixTable),
            keep_star=bool,
            left_aligned=bool,
            vep_root=str)
-def split_multi_hts(ds, keep_star=False, left_aligned=False, vep_root='vep') -> MatrixTable:
+def split_multi_hts(ds, keep_star=False, left_aligned=False, vep_root='vep') -> Union[Table, MatrixTable]:
```
Comment (tpoterba) [inline python/hail/methods/statgen.py]: don't generate Unions - this causes problems in IDEs. The right way to do it is with generics, but that's kind of a pain.

Comment (konradjk) [reply python/hail/methods/statgen.py]: I just copied what was in split_multi. What do you propose changing both to?

Comment (konradjk) [reply python/hail/methods/statgen.py]: Also my IDE works totally fine with Unions (better than not adding a rtype)

Comment (tpoterba) [reply python/hail/methods/statgen.py]: just delete the return type hints. If we're maximizing IDE usefulness, then this is better than warnings everywhere that "`Union[Table, MatrixTable]` has no attribute 'annotate_rows'"`

Comment (tpoterba) [reply python/hail/methods/statgen.py]: alternatively, we use generics. it's not that hard, and isn't so bad here.

Comment (tpoterba) [reply python/hail/methods/statgen.py]: it just gets REAL weird when we think about applying it to the expression system.

**Pull request merged**